### PR TITLE
sorter: fix unified sorter magic numbers & reduce memory consumption by channels

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- Unexpectedly large memory consumption by Unified Sorter when tables are many.
- Too many magic numbers in Unified Sorter.

### What is changed and how it works?
- Reduce heapSorter inputCh size
- Add constants for some magic numbers used for sizes to better manage these sizes.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note

- Fix Unified Sorter memory consumption when tables are many.

### CODE REVIEW

**Review by ben1009 — APPROVED** [review]

File: cdc/puller/sorter/heap_sorter.go [hunk 1]
```diff
@@ -85,7 +86,7 @@ type heapSorter struct {
 func newHeapSorter(id int, out chan *flushTask) *heapSorter {
 	return &heapSorter{
 		id:        id,
-		inputCh:   make(chan *model.PolymorphicEvent, 1024*1024),
+		inputCh:   make(chan *model.PolymorphicEvent, sortHeapInputChSize),
```
Comment (leoppro) [inline cdc/puller/sorter/heap_sorter.go:89]:
```suggestion
		inputCh:   make(chan *model.PolymorphicEvent, sortHeapInputChSize),
```
the `sortHeapInputChSize` is not 1024*1024, is it expected?

**Review by overvenus — APPROVED** [review]
Could you paste some ticdc metrics graphes that shows how these changes reduces memory consumption?

**Review by leoppro — APPROVED** [review]

File: cdc/puller/sorter/heap_sorter.go (hunk 1 above)
Comment (liuzix) [reply cdc/puller/sorter/heap_sorter.go:89]: Yes. 1024 * 1024 is too large.

File: pkg/config/config.go [hunk 2]
```diff
@@ -158,9 +158,9 @@ var defaultServerConfig = &ServerConfig{
 	ProcessorFlushInterval: TomlDuration(100 * time.Millisecond),
 	Sorter: &SorterConfig{
 		NumConcurrentWorker:    4,
-		ChunkSizeLimit:         1024 * 1024 * 1024, // 1GB
-		MaxMemoryPressure:      80,
-		MaxMemoryConsumption:   8 * 1024 * 1024 * 1024, // 8GB
+		ChunkSizeLimit:         128 * 1024 * 1024,       // 128MB
+		MaxMemoryPressure:      30,                      // 30% is safe on machines with memory capacity <= 16GB
+		MaxMemoryConsumption:   16 * 1024 * 1024 * 1024, // 16GB
```
Comment (overvenus) [inline pkg/config/config.go:163]: > // the maximum memory consumption allowed for in-memory sorting
> MaxMemoryConsumption uint64 `toml:"max-memory-consumption" json:"max-memory-consumption"`

Is it safe for machines with 16G RAM?

**Pull request merged**